configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml

```diff
@@ -105,22 +105,143 @@
 <script type="text/javascript">
     jq(function() {
 
-      jq("#monthsDispensed input").change(function() {
-        var monthsDispensed= htmlForm.getValueIfLegal('monthsDispensed.value');
-          if (typeof monthsDispensed !== "undefined" &amp;&amp; monthsDispensed !== null &amp;&amp; (monthsDispensed.length > 0)) {
+        var dispensedToChwConceptId = <lookup expression="fn.getConcept('CIEL:1555').id"/>;
+        var dispensedToPatientConceptId = <lookup expression="fn.getConcept('CIEL:162571').id"/>;
+
+
+        var dispensedToValue= getValue('dispensedTo.value');
+        if (dispensedToValue == dispensedToPatientConceptId) {
+            getField('chwName.value').fadeTo(250,0.33);
+            getField('chwName.value').prop( 'disabled', true );
+        } else if (dispensedToValue == dispensedToChwConceptId) {
+            getField('dispensedInCommunity.value').prop( 'disabled', true );
+        }
+
+        var validateNextDispensingDate = function() {
+            var today = new Date();
+            var currentEncounterDate = '<lookup expression="encounter.getEncounterDatetime().getTime()"/>';
+            if (typeof currentEncounterDate !== "undefined" &amp;&amp; currentEncounterDate !== null &amp;&amp; (currentEncounterDate.length > 0)) {
+              today = new Date(+currentEncounterDate);
```

Comment:
Should this variable be renamed to currentEncounterDate?  If the encounter is in the past, then "today" is no longer "today"?